pkg/manager/member/tidb_member_manager.go

```diff
@@ -1044,6 +1053,86 @@ func (m *tidbMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, set
 	return nil
 }
 
+func (m *tidbMemberManager) setServerLabels(tc *v1alpha1.TidbCluster) (int, error) {
+	if m.deps.NodeLister == nil {
+		klog.V(4).Infof("Node lister is unavailable, skip setting store labels for TiKV of TiDB cluster %s/%s. This may be caused by no relevant permissions", tc.Namespace, tc.Name)
+		return 0, nil
+	}
+
+	ns := tc.GetNamespace()
+	// for unit test
+	setCount := 0
+
+	pdCli := controller.GetPDClient(m.deps.PDControl, tc)
+	config, err := pdCli.GetConfig()
+	if err != nil {
+		return setCount, err
+	}
+
+	var zoneLabel string
+	for _, l := range config.Replication.LocationLabels {
+		if l == topologyZoneLabel {
+			zoneLabel = l
+		} else if l == tidbDCLabel {
+			zoneLabel = l
+		}
+	}
+	if zoneLabel == "" {
+		klog.Infof("zone labels not found in pd location-labels %v, skip set labels", config.Replication.LocationLabels)
+		return 0, nil
+	}
+
+	for name, db := range tc.Status.TiDB.Members {
+		if !db.Health {
+			continue
+		}
+		ordinal, err := parserOrdinal(name)
+		if err != nil {
+			return setCount, err
+		}
+
+		pod, err := m.deps.PodLister.Pods(ns).Get(name)
+		if err != nil {
+			return setCount, fmt.Errorf("setServerLabels: failed to get pods %s for cluster %s/%s, error: %s", name, ns, tc.GetName(), err)
+		}
+
+		nodeName := pod.Spec.NodeName
+		labels, err := getNodeLabels(m.deps.NodeLister, nodeName, config.Replication.LocationLabels)
+		if err != nil || len(labels) == 0 {
+			klog.Warningf("node: [%s] has no node labels, skipping set store labels for Pod: [%s/%s]", nodeName, ns, name)
+			continue
+		}
+		// add the special `zone` label because tidb depends on this label for follower read.
+		labels[tidbDCLabel] = labels[zoneLabel]
+
+		if err := m.deps.TiDBControl.SetServerLabels(tc, ordinal, labels); err != nil {
+			klog.Warningf("cluster %s/%s set server labels for pod %s failed, error: %v", ns, tc.GetName(), name, err)
+			continue
+		}
```

Comment:
If tidb version is old, does it always log warnnings?

Reply:
I have added a min tidb version check at the beginning of this method

Reply:
maybe use the `pkg/util/cmpver` pkg to compare version, it works for dirty versions or the latest version.

Reply:
For example:
https://github.com/pingcap/tidb-operator/blob/00002ede5e5b2c811d8bbff4043a4cc7a2a1e2a6/pkg/manager/member/tiflash_util.go#L109-L115

Reply:
updated